src/main/java/io/blamer/hub/pg/TokenValidated.java

```diff
@@ -0,0 +1,57 @@
+package io.blamer.hub.pg;
+
+import io.blamer.hub.exceptions.TokenAlreadyExists;
+import io.blamer.hub.model.Token;
+import io.blamer.hub.model.Tokens;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.r2dbc.core.DatabaseClient;
+import org.springframework.stereotype.Component;
+import reactor.core.publisher.Mono;
+@Component
+public class TokenValidated implements Tokens {
+    /**
+     * Tokens.
+     */
+    private final Tokens tokens;
+    /**
+     * Database Client.
+     */
+    private final DatabaseClient db;
+
+    /**
+     * Ctor.
+     *
+     * @param tks Tokens
+     * @param dbc  Database Client
+     */
+    public TokenValidated(
+        @Qualifier("pgTokens") final Tokens tks,
+        final DatabaseClient dbc
+    ) {
+        this.tokens = tks;
+        this.db = dbc;
+    }
+
+    @Override
+    public Mono<Void> add(final Token token) {
+        return validateToken(token)
```

Comment:
@iColdLight 
This logic can be decorated, presented as at least 2 classes: first where you just insert token, and the second one where you check token validity and integrity, and then insert into database (reuse first object)
What do you think?